hs_core/management/commands/fix_resource_versioning.py

```diff
@@ -0,0 +1,117 @@
+                        print(f'{res_id} is made a copy of {copy_of_res_id}', flush=True)
+                        fixed = True
+                elif not row[2]:
+                    # isVersionOf is not empty but isReplacedBy is empty
+                    version_res_id = row[0]
+                    try:
+                        version_res = get_resource_by_shortkey(version_res_id)
+                    except Exception:
+                        print(f'Resource {version_res_id} does not exist', flush=True)
+                        continue
+                    hs_identifier = version_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
+                    if res.metadata.relations.all().filter(type='isVersionOf').exists():
+                        res.metadata.relations.all().filter(type='isVersionOf').all().delete()
+                    res.metadata.create_element('relation', type='isVersionOf', value=hs_identifier.url)
+                    resource_modified(res, overwrite_bag=False)
+                    fixed = True
+                elif not row[0]:
+                    # isVersionOf is empty but isReplacedBy is not empty
+                    replace_by_res_id = row[2]
+                    try:
+                        replace_by_res = get_resource_by_shortkey(replace_by_res_id)
+                    except Exception:
+                        print(f'Resource {replace_by_res_id} does not exist', flush=True)
+                        continue
+                    hs_identifier = replace_by_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
+                    if res.metadata.relations.all().filter(type='isReplacedBy').exists():
+                        res.metadata.relations.all().filter(type='isReplacedBy').all().delete()
+                    res.metadata.create_element('relation', type='isReplacedBy', value=hs_identifier.url)
+                    resource_modified(res, overwrite_bag=False)
+                    fixed = True
+                else:
+                    # both isReplacedBy and isVersionOf are not empty - need to only keep this pair of relationship
+                    # and delete other pairs to meet the single obsolecense chain requirement
+                    version_res_id = row[0]
+                    try:
+                        version_res = get_resource_by_shortkey(version_res_id)
+                    except Exception:
+                        print(f'Resource {version_res_id} does not exist', flush=True)
+                        continue
+                    replace_res_id = row[2]
+                    try:
+                        replace_res = get_resource_by_shortkey(replace_res_id)
+                    except Exception:
+                        print(f'Resource {replace_res_id} does not exist', flush=True)
+                        continue
+                    if res.metadata.relations.all().filter(type='isVersionOf').exists():
```

Comment:
I need this if statement so that resource modified can be applied and the counter for the number of fixed resources can be updated only when the relation exists.

Reply:
Never mind. I was thinking of a different statement. Yes, agreed this if statement here is not needed.